config/shared_config.go

```diff
@@ -59,6 +59,10 @@ type RaftConf struct {
 	// pass between heartbeats. That is, a leader sends heartbeat
 	// messages to maintain its leadership every HeartbeatTick ticks.
 	HeartbeatTicks uint32
+	// Limits the max number of in-flight blocks (i.e. Raft messages).
+	MaxInflightBlocks uint32
```

Comment:
Does the proposer wait to cut a block once this limit is reached?

Reply:
This is an internal raft parameter, if the number of in flight messages is beyond this, the leader will stop proposing. I think it will lead to the proposing go-routine to block. 

If the propose go-routine is blocked, it will not pull blocks from the channel that connects the replication and the block creator, which will eventually cause the block creator to stop. But all that in the next PR, not now...